splash/qtutils.py

```diff
@@ -217,3 +221,125 @@ def connect(self, func, **kwargs):
     def disconnect(self, callback_id):
         cb = self.callbacks.pop(callback_id)
         self.signal.disconnect(cb)
+
+
+# Brain-dead simple endianness detection needed because QImage stores
+# bytes in host-native order.
+#
+# XXX: do we care about big endian hosts?
+# XXX: is there a better way?
+_is_little_endian = struct.pack('<i', 0x1122) == struct.pack('=i', 0x1122)
+
+
+def qimage_to_pil_image(qimage):
+    """Convert QImage (in ARGB32 format) to PIL.Image (in RGBA mode)."""
+    # In our case QImage uses 0xAARRGGBB format stored in host endian order,
+    # we must convert it to [0xRR, 0xGG, 0xBB, 0xAA] sequences used by pillow.
+    buf = qimage.bits().asstring(qimage.numBytes())
+    if not _is_little_endian:
+        # If host is big-endian, convert endianness first.
+        buf = swap_byte_order_i32(buf)
+    # QImage's 0xARGB in little-endian becomes [0xB, 0xG, 0xR, 0xA] for pillow,
+    # hence the 'BGRA' decoder argument.
+    return Image.frombytes(
+        'RGBA', (qimage.size().width(), qimage.size().height()), buf,
+        'raw', 'BGRA')
+
+
+def pil_image_to_qimage(pil_image):
```

Comment:
This function is unused; even though it may be useful in some cases, I think it is better to delete it from Splash.